### PR TITLE
Adds documentation of the ROS integration options.

### PR DESCRIPTION
**Pull request by wohe**

### CODE REVIEW

File: docs/source/options.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+   limitations under the License.
+
+=======================
+Configuration Reference
+=======================
+
+Top-level options of the Cartographer ROS integration:
+
+map_frame
+  The ROS frame id to use for publishing submaps, the parent frame of poses,
```
Comment (damonkohler) [inline docs/source/options.rst]: s/id/ID

File: docs/source/options.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+=======================
+Configuration Reference
+=======================
+
+Top-level options of the Cartographer ROS integration:
```
Comment (damonkohler) [inline docs/source/options.rst]: Cartographer's ROS integration top-level options:

Add some text about Lua configuration? All values must be specified, etc.

File: docs/source/options.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+map_frame
+  The ROS frame id to use for publishing submaps, the parent frame of poses,
+  usually "map".
+
+tracking_frame
+  The ROS frame id of the frame that is tracked by the SLAM algorithm. If an IMU
+  is used it should be at its position, although it might be rotated. A common
```
Comment (damonkohler) [inline docs/source/options.rst]: If an IMU is used, ...

File: docs/source/options.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  choice is "base_link".
+
+published_frame
+  The ROS frame id to use as the child frame for publishing poses. For example
+  "odom" if an "odom" frame is supplied by a different part of the system. In
+  this case the pose of "odom" in the map frame is to be published.
```
Comment (damonkohler) [inline docs/source/options.rst]: ... will be published.

Add a second example for when odom is not published by another node.

File: docs/source/options.rst [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+odom_frame
+  Only used if provide_odom_frame is true. The frame between published_frame and 
+  map_frame to be used for publishing the (non-loop-closed) local SLAM result.
+  Usually "odom".
+
+provide_odom_frame
+  If enabled, not only the global SLAM result is published to place to
```
Comment (damonkohler) [inline docs/source/options.rst]: If enabled, the local, non-loop-closed, continuous pose will be published as the odom_frame.

File: docs/source/options.rst [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  published_frame inside the map_frame, but also the local SLAM result at the
+  odom_frame in between the two.
+
+use_odometry_data
+  If enabled, subscribes to `nav_msgs::Odometry`_ under the topic "odom".
```
Comment (damonkohler) [inline docs/source/options.rst]: on the topic

File: docs/source/options.rst [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  Odometry must be provided in this case, and the information will be included
```
Comment (damonkohler) [inline docs/source/options.rst]: Add a link to nav_msgs/Odometry

File: docs/source/options.rst [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  in SLAM.
+
+use_constant_odometry_variance
+  If enabled, the configured variances will be used instead of the ones provided
+  in the Odometry messages.
```
Comment (damonkohler) [inline docs/source/options.rst]: nav_msgs/Odometry

File: docs/source/options.rst [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  The variance to use for the translational component of odometry to use if
+  use_constant_odometry_variance is enabled.
+
+constant_odometry_rotational_variance
+  The variance to use for the rotational component of odometry to use if
+  use_constant_odometry_variance is enabled.
+
+publish_occupancy_grid
+  If enabled, a background thread will continuously compute and publish
+  `nav_msgs::OccupancyGrid`_ messages under the "map" topic. Depending on the
```
Comment (damonkohler) [inline docs/source/options.rst]: on the topic

Fix this everywhere.

File: docs/source/options.rst [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+  detected.
+
+num_lasers_3d
+  Number of 3D lasers to subscribe to. Must be a positive value if and only if
+  using 3D SLAM. Subscribes to `sensor_msgs::PointCloud2`_ under the "points2"
+  topic for one laser, or topics "points2_1", "points2_2", etc for multiple
+  lasers.
+
+lookup_transform_timeout_sec
+  Timeout in seconds to use for looking up transforms using tf2.
```
Comment (damonkohler) [inline docs/source/options.rst]: tf2 link

File: docs/source/options.rst [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+published_frame
+  The ROS frame ID to use as the child frame for publishing poses. For example
+  "odom" if an "odom" frame is supplied by a different part of the system. In
+  this case the pose of "odom" in the *map_frame* will be published. Otherwise,
+  setting it to "base_link" is likely appropriate.
+
+odom_frame
+  Only used if *provide_odom_frame* is true. The frame between *published_frame*
+  and *map_frame* to be used for publishing the (non-loop-closed) local SLAM
+  result. Usually "odom".
```
Comment (rohbotics) [inline docs/source/options.rst]: Specify if the topic name is in private namespace or not using "~/odom" or "/odom".

Do this on all the topic names please.

**Pull request merged**